src/main/java/uk/gov/ons/ctp/response/action/message/impl/CaseNotificationReceiverImpl.java

```diff
@@ -18,12 +22,33 @@
 
   @Autowired private CaseNotificationService caseNotificationService;
 
+  @Qualifier("genericRabbitTemplate")
+  @Autowired
+  private RabbitTemplate rabbitTemplate;
+
+  @Autowired private ObjectMapper objectMapper;
+
+  private static final String ACTION_OUTBOUND_EXCHANGE = "action-outbound-exchange";
+  private static final String ACTION_ROUTING_KEY = "Action.CaseNotificationHandled.binding";
+
   @Override
   @ServiceActivator(
       inputChannel = "caseNotificationTransformed",
       adviceChain = "caseNotificationRetryAdvice")
   public void acceptNotification(final CaseNotification caseNotification) throws CTPException {
-    log.debug("Receiving case notification for case id {}", caseNotification.getCaseId());
-    caseNotificationService.acceptNotification(caseNotification);
+
+    try {
+      log.debug("Receiving case notification for case id {}", caseNotification.getCaseId());
+      caseNotificationService.acceptNotification(caseNotification);
+    } finally {
+      try {
+        rabbitTemplate.convertAndSend(
```

Comment:
Why is this message needed and what value are we getting from publishing this message?

Reply:
The purpose is to know when the processing of the case notification has completed, whether it succeeds or fails (hence it's in a finally block).  The other methods we looked at (polling the database for case events etc) were unreliable if the operation failed i.e. the integration test would hang.  Currently the ActionEndpointIT  (line 145) uses this message to know when to continue processing. We had a race condition before we implemented this.

Reply:
Fair enough misread the test that you needed to wait on that for the next rest request